torch/distributed/_tensor/_dispatch.py

```diff
@@ -231,6 +231,10 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             if output_sharding.output_spec is not None:
                 return args[0]
             else:
+                if op_call == aten._amp_foreach_non_finite_check_and_unscale_.default:
+                    found_inf = args[1]
+                    work = mesh.get_group().allreduce(found_inf)  # type: ignore[call-overload]
```

Comment:
this would only work for 1-d mesh

Reply:
More context, `get_group()` without argument would error out if mesh is 2D.

Reply:
not a blocker for this PR, but is there a recommended approach for 2d+  @XilunWu  @wz337 ? for-loop over mesh dims and all-reduce over `placement=shard` and ignore `placement=replicate` ?

Reply:
probably could just make this `found_inf` into a `DTensor` with the right placements and then call `full_tensor`